src/esp/sensor/CameraSensor.cpp

```diff
@@ -14,62 +14,81 @@
 namespace esp {
 namespace sensor {
 
+CameraSensorSpec::CameraSensorSpec()
+    : VisualSensorSpec(), channels(4), observationSpace("") {
+  uuid = "rgba_camera";
+  sensorSubType = SensorSubType::Pinhole;
+}
+
+void CameraSensorSpec::sanityCheck() {
+  VisualSensorSpec::sanityCheck();
+  CORRADE_ASSERT(sensorSubType == SensorSubType::Pinhole ||
+                     sensorSubType == SensorSubType::Orthographic,
+                 "CameraSensor::CameraSensorSpec(): sensorSpec does not have "
+                 "SensorSubType "
+                 "Pinhole or Orthographic", );
+}
+
+bool CameraSensorSpec::operator==(const CameraSensorSpec& a) {
+  return VisualSensorSpec::operator==(a) && channels == a.channels &&
+         observationSpace == a.observationSpace;
+}
+
 CameraSensor::CameraSensor(scene::SceneNode& cameraNode,
-                           const SensorSpec::ptr& spec)
+                           const CameraSensorSpec::ptr& spec)
     : VisualSensor(cameraNode, spec),
       baseProjMatrix_(Magnum::Math::IdentityInit),
       zoomMatrix_(Magnum::Math::IdentityInit) {
+  // Sanity check
+  cameraSensorSpec_->sanityCheck();
```

Comment:
What if cameraSensorSpec_ is nullptr? Then it would segfault here.
You should have a CORRADE_ASSERT and when it happens, give the error saying "the input spec is illegal."

It can occur. For example:

externally:
```
std::shared_ptr<SensorSpec> sp = make_shared<SensorSpec>(...);
std::shared_ptr<CameraSensorSpec> bad_csp = std::static_pointer_cast<CameraSensorSpec>(sp); // CAREFUL: This is already BAD! but bad_csp will NOT be nullptr so far.
// Then someone uses this bad_csp to construct the CameraSensor, and then cameraSensorSpec_ would be nullptr because of the dynamic_pointer_cast.